### PR TITLE
getting-started/bookshelf: add integration tests

### PR DESCRIPTION
**Pull request by broady**

### CODE REVIEW

File: internal/testutil/runmain.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package testutil
```
Comment (adg) [inline internal/testutil/runmain.go]: // Package testutil ...

File: internal/webtest/webtest.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package webtest
```
Comment (adg) [inline internal/webtest/webtest.go]: // Package webtest...

File: internal/webtest/webtest.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+)
+
+type W struct {
```
Comment (adg) [inline internal/webtest/webtest.go]: need docs

File: internal/testutil/runmain.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+)
+
+func BuildMain(t *testing.T) *Runner {
```
Comment (adg) [inline internal/testutil/runmain.go]: need docs

File: internal/webtest/webtest.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+	for time.Now().Before(deadline) {
+		conn, err := net.Dial("tcp", w.host)
+		if err != nil {
+			time.Sleep(retryDelay)
+			continue
+		}
+		conn.Close()
+		break
+	}
```
Comment (adg) [inline internal/webtest/webtest.go]: shouldn't something happen if the timeout is hit without net being ready? t.fatal?

File: internal/testutil/runmain.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+	}()
+
+	// Try to gracefully kill the process.
+	if err := cmd.Process.Signal(syscall.SIGINT); err != nil {
+		r.t.Error(err)
+	}
+
+	select {
+	case <-time.After(5 * time.Second):
+		r.t.Error("Timed out with SIGINT, hard killing.")
```
Comment (adg) [inline internal/testutil/runmain.go]: s/hard killing/trying SIGKILL/

File: internal/testutil/runmain.go (hunk 1 above)
Comment (broady) [inline internal/testutil/runmain.go]: Already there.

`// Package testutil provides test helpers for the golang-samples repo.`

File: internal/webtest/webtest.go (hunk 3 above)
Comment (broady) [inline internal/webtest/webtest.go]: Done

File: internal/testutil/runmain.go (hunk 6 above)
Comment (broady) [inline internal/testutil/runmain.go]: Done

File: internal/webtest/webtest.go (hunk 5 above)
Comment (broady) [inline internal/webtest/webtest.go]: t.Fatal now

File: internal/testutil/runmain.go (hunk 4 above)
Comment (broady) [inline internal/testutil/runmain.go]: Done

File: internal/webtest/webtest.go (hunk 2 above)
Comment (broady) [inline internal/webtest/webtest.go]: Done

File: getting-started/bookshelf/app/app.go [hunk 7]
```diff
@@ -35,6 +36,11 @@ var (
 )
 
 func main() {
+	addHandlers()
```
Comment (adg) [inline getting-started/bookshelf/app/app.go]: registerHandlers?

Comment (broady) [inline getting-started/bookshelf/app/app.go]: Done

**Comment by adg** [comment]
LGTM

**Pull request merged**